pdns/sha.hh

```diff
@@ -51,3 +52,64 @@ inline std::string pdns_sha512sum(const std::string& input)
   SHA512(reinterpret_cast<const unsigned char*>(input.c_str()), input.length(), result);
   return std::string(result, result + sizeof result);
 }
+
+namespace pdns
+{
+class SHADigest
```

Comment:
Maybe you could disable default constructor ?

Reply:
Default constructor is implicitly deleted if another one exists I think